mlflow/evaluation/__init__.py

```diff
@@ -1,4 +1,12 @@
+from mlflow.entities.assessment_source import AssessmentSource, AssessmentSourceType
```

Comment:
The Databricks Mosaic AI Agent framework depends on these entities being part of the `mlflow.evaluation` module. Hence, we import them here. I also added a test case for this: `test_evaluation_module_exposes_relevant_apis_for_logging`